.github/workflows/ubuntu-ci-x86_64-intel.yaml

```diff
@@ -115,6 +115,14 @@ jobs:
           echo "    - spec: qt@5.15.3" >> ${SPACK_SYSTEM_CONFIG_PATH}/packages.yaml
           echo "      prefix: /usr" >> ${SPACK_SYSTEM_CONFIG_PATH}/packages.yaml
 
+          # Add external grep.
+          echo "" >> ${SPACK_SYSTEM_CONFIG_PATH}/packages.yaml
+          echo "  grep:" >> ${SPACK_SYSTEM_CONFIG_PATH}/packages.yaml
+          echo "    buildable: False" >> ${SPACK_SYSTEM_CONFIG_PATH}/packages.yaml
+          echo "    externals:" >> ${SPACK_SYSTEM_CONFIG_PATH}/packages.yaml
+          echo "    - spec: grep@3.11" >> ${SPACK_SYSTEM_CONFIG_PATH}/packages.yaml
+          echo "      prefix: /usr" >> ${SPACK_SYSTEM_CONFIG_PATH}/packages.yaml
```

Comment:
This was added to fix the currently broken intel build. See the discussion linked in my response to Dom. This PR isn't the cause of the failure but I've included the fix here to unblock this PR and others.

Reply:
I wonder if we should make this a "default" for all sites that we maintain:
- Check if spack has the necessary information to do `spack external find grep`
- Add this command to all GitHub workflows
- Update documentation (NewSiteConfig) to include the external find grep
- Add external grep spec for all containers and all preconfigured sites (tier-1; tier-2 is up to the individuals maintaining those sites)

Reply:
hmm, that's an interesting idea. For reasons I don't fully understand this only manifests when building with intel and gcc-11 which is the case in this system and in the ubuntu image. Intel plus gcc-12 or gcc-13 and grep is fine (but other things break). Good catch on 'exteral find' which definitely was the correct call here.

I think without more info on where/why this happens I'd be hesitant to make this default everywhere even though it's probably not harmful.

Reply:
Alright - the change I added using `external find grep` doesn't work and we have to manually modify the packages file. This is making the intel packages change more interesting since grep is definitely available everywhere